misc/formatter/modules/formatter-cli/src/main/java/org/ballerinalang/formatter/cli/FormatUtil.java

```diff
@@ -0,0 +1,358 @@
+                            outStream.println(System.lineSeparator() + Messages.getSuccessMessage());
+                        } else {
+                            outStream.println(Messages.getFilesToModify() + System.lineSeparator() + ballerinaFilePath);
+                        }
+                    } else {
+                        outStream.println(Messages.getNoChanges());
+                    }
+                } else if (Paths.get(argList.get(0)).toFile().isFile()) {
+                    // If file is a regular file but not a ballerina source file
+                    // throw the following exception.
+                    throw LauncherUtils.createLauncherException(Messages.getNotABallerinaFile());
+                } else {
+                    moduleName = argList.get(0);
+
+                    // Check whether the module dir exists.
+                    if (!FormatUtil.isModuleExist(moduleName, sourceRootPath)) {
+                        // If module directory doesn't exist and contains a "."
+                        // throw a exception to say file or module doesn't exist.
+                        // Else throw a exception to say module doesn't exist.
+                        if (moduleName.contains(".")) {
+                            throw LauncherUtils.createLauncherException(Messages
+                                    .getNoBallerinaModuleOrFile(moduleName));
+                        } else {
+                            throw LauncherUtils.createLauncherException(Messages.getNoModuleFound(moduleName));
+                        }
+                    }
+
+                    // Check whether the given directory is not in a ballerina project.
+                    if (FormatUtil.notABallerinaProject(sourceRootPath)) {
+                        throw LauncherUtils.createLauncherException(Messages.getNotBallerinaProject());
+                    }
+                    BLangPackage bLangPackage = FormatUtil
+                            .compileModule(sourceRootPath, getModuleName(moduleName));
+
+                    // Iterate and format the ballerina package.
+                    List<String> formattedFiles = iterateAndFormat(bLangPackage, sourceRootPath, dryRun);
+                    generateChangeReport(formattedFiles, dryRun);
+                }
+            } else {
+                List<BLangPackage> packages = FormatUtil.compileProject(sourceRootPath);
+                List<String> formattedFiles = new ArrayList<>();
+                // Iterate and format all the ballerina packages.
+                for (BLangPackage bLangPackage : packages) {
+                    formattedFiles.addAll(iterateAndFormat(bLangPackage, sourceRootPath, dryRun));
+                }
+
+                generateChangeReport(formattedFiles, dryRun);
+            }
+        } catch (IOException | NullPointerException e) {
+            throw LauncherUtils.createLauncherException(Messages.getException());
```

Comment:
Shall we append the original exception `e` as well? That would  give more information on the error.

Reply:
Done